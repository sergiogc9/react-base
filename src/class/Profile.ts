export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
