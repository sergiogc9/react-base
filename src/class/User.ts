export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
