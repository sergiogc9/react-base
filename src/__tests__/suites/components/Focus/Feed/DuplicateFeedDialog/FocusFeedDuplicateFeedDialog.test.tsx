import React from "react";
import { shallow, mount } from "enzyme";

import { FocusObject } from "@src/class/Focus";

import FocusFeedDuplicateFeedDialog from "@src/components/Focus/Feed/DuplicateFeedDialog/FocusFeedDuplicateFeedDialog";

const focus: FocusObject = {
	deleted_at: new Date(1549287550563),
	feeds: { online: [], socialmedia: [], print: [] },
	id: "focus-id-1",
	inserted_at: new Date(1549287550563),
	name: "focus-name-1",
	updated_at: new Date(1549287550563),
	url_logo: "http://fake-url.com",
	acl_users: [
		{
			user_id: 'fake-user-id-1',
			discover_role: 'viewer',
			last_name: "Name",
			first_name: "User"
		},
		{
			user_id: 'fake-user-id-2',
			discover_role: 'viewer',
			last_name: 'User 2',
			first_name: 'Name 2'
		}
	]
};

const focus2: FocusObject = {
	...focus,
	id: "focus-id-2",
	name: "focus-name-2",
};

const focusList: FocusObject[] = [focus, focus2];

describe("FocusFeedDuplicateFeedDialog", () => {
	let wrapper: any;
	let onDuplicateFeedStart = jest.fn();
	let onToggleShowDuplicateFeed = jest.fn();

	const getComponent = () => <FocusFeedDuplicateFeedDialog
		focusList={focusList}
		focusId="focus-id"
		feedName="feed-name"
		loadingDuplicatedFeed={false}
		duplicatedFeedId={null}
		onDuplicateFeedStart={onDuplicateFeedStart}
		onToggleShowDuplicateFeed={onToggleShowDuplicateFeed}
	/>;

	beforeEach(() => {
		onDuplicateFeedStart = jest.fn();
		onToggleShowDuplicateFeed = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with focusList null", () => {
		wrapper.setProps({ focusList: null });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with loading duplicated feed", () => {
		wrapper.setProps({ loadingDuplicatedFeed: true });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with duplicated feed id", () => {
		wrapper.setProps({ duplicatedFeedId: "dup-feed-id" });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component feed name change event", () => {
		wrapper = mount(getComponent());
		wrapper.find('#focusFeedDuplicateFeedDialogFeedName').at(0).props().onChange("new-feed-name");
		expect(wrapper.state().editedFeedName).toEqual("new-feed-name");
	});

	it("Component focus select change event", () => {
		wrapper = mount(getComponent());
		wrapper.find('#focusFeedDuplicateFeedDialogFocusSelect').at(0).props().onChange("selected-focus-id");
		expect(wrapper.state().selectedFocusId).toEqual("selected-focus-id");
	});

	it("Component focus cancel button handler", () => {
		wrapper = mount(getComponent());
		wrapper.find('#launchmetricsDialogCancelBtn').at(0).simulate('click');
		expect(onToggleShowDuplicateFeed).toHaveBeenCalledTimes(1);
	});

	it("Component focus accept button handler to start duplicating the feed", () => {
		wrapper = mount(getComponent());
		wrapper.find('#launchmetricsDialogAcceptBtn').at(0).simulate('click');
		expect(onDuplicateFeedStart).toHaveBeenCalledTimes(1);
	});

	it("Component focus accept button handler to close dialog", () => {
		wrapper = mount(getComponent());
		wrapper.setProps({ duplicatedFeedId: "dup-feed-id" });
		wrapper.find('#launchmetricsDialogAcceptBtn').at(0).simulate('click');
		expect(onToggleShowDuplicateFeed).toHaveBeenCalledTimes(1);
	});

});
