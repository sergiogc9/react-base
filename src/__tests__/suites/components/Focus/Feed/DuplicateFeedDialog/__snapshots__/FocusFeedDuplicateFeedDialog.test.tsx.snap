// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusFeedDuplicateFeedDialog Component snapshot 1`] = `"<div id=\\"focusFeedDuplicateFeed\\"><span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"focusFeedDuplicateFeedDialog\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">feed.form.duplicate.dialog.title</div></div><div class=\\"dialog-content\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input\\"><label for=\\"focusFeedDuplicateFeedDialogFeedName\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">feed.form.duplicate.feed_name.label</label><input type=\\"text\\" value=\\"feed.form.duplicate.feed_name.valuefeed-name\\" id=\\"focusFeedDuplicateFeedDialogFeedName\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"/></div><div id=\\"focusFeedDuplicateFeedDialogFocusSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedDuplicateFeedDialogFocusSelect\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">focus.value</label><div id=\\"focusFeedDuplicateFeedDialogFocusSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedDuplicateFeedDialogFocusSelect\\" value=\\"focus-id\\"/></div></div></div></div><div class=\\"dialog-footer md-grid \\"><button id=\\"launchmetricsDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.cancel</button><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.accept</button></div></div></section></div></span></div>"`;

exports[`FocusFeedDuplicateFeedDialog Component snapshot with duplicated feed id 1`] = `"<div id=\\"focusFeedDuplicateFeed\\"><span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"focusFeedDuplicateFeedDialog\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">feed.form.duplicate.dialog.title</div></div><div class=\\"dialog-content\\"><div>feed.form.duplicate.dupped_ok</div><div><a href=\\"/focus/focus-id/feed/dup-feed-id\\">feed.form.duplicate.go_to_new_feed</a></div></div><div class=\\"dialog-footer md-grid center-buttons\\"><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.accept</button></div></div></section></div></span></div>"`;

exports[`FocusFeedDuplicateFeedDialog Component snapshot with focusList null 1`] = `null`;

exports[`FocusFeedDuplicateFeedDialog Component snapshot with loading duplicated feed 1`] = `"<div id=\\"focusFeedDuplicateFeed\\"><span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"focusFeedDuplicateFeedDialog\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">feed.form.duplicate.dialog.title</div></div><div class=\\"dialog-content\\"><svg id=\\"focusFeedDuplicateFeedDialogLoading\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate\\" width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg></div><div class=\\"dialog-footer md-grid \\"><button id=\\"launchmetricsDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.cancel</button><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.accept</button></div></div></section></div></span></div>"`;
